libioc/helpers.py

```diff
@@ -532,35 +535,27 @@ def exec_passthru(
     return None, None, child.returncode
 
 
-# ToDo: replace with (u)mount library
-def umount_command(
-    mountpoint: typing.Optional[typing.Union[
-        libioc.Types.AbsolutePath,
-        typing.List[libioc.Types.AbsolutePath],
-    ]]=None,
-    options: typing.Optional[typing.List[str]]=None,
-    force: bool=False,
-    ignore_error: bool=False,
-    logger: typing.Optional['libioc.Logger.Logger']=None
-) -> typing.List[str]:
-    """Return the command to unmount a mountpoint."""
-    cmd = ["/sbin/umount"]
-
-    if force is True:
-        cmd.append("-f")
-
-    if options is not None and len(options) != 0:
-        cmd + options
-
-    if isinstance(mountpoint, list):
-        cmd += mountpoint
-    elif isinstance(mountpoint, libioc.Types.AbsolutePath):
-        cmd.append(str(mountpoint))
-
-    if ignore_error is True:
-        cmd.append(">/dev/null 2>&1 || :")
-
-    return cmd
+def mount(
+    destination: str,
+    source: str=None,
+    fstype: str="nullfs",
+    opts: typing.List[str]=[]
+) -> None:
+    """Mount a filesystem using libc."""
+    data = dict(
+        fstype=fstype,
+        fspath=destination
+    )
+    if source is not None:
+        data["target"] = source
+    for opt in opts:
+        data[opt] = None
+    jiov = libjail.Jiov(data)
+    if libjail.dll.nmount(jiov.pointer, len(jiov), 0) != 0:
```

Comment:
i understand that you're basically just using py-jail as libc interface, but it feels dirty 😅
maybe write that in a function's comment

Reply:
> i understand that you're basically just using py-jail as libc interface, but it feels dirty 😅

Yes, it does. The reason to re-use the DLL here is that it was already loaded by py-jail (https://github.com/gronke/py-jail/blob/master/jail/libc.py). It would have been possible to use the dll from py-freebsd_sysctl (https://github.com/gronke/py-freebsd_sysctl/blob/master/freebsd_sysctl/libc.py), but in either case the result is the same.

Although this are just 3 LOC, I didn't want to repeat them a third time in libioc. It's a public interface of either py-jail or py-freebsd_sysctl, so that it should not make a difference - even when calling nmount, which is totally unrelated to both (jail and sysctl).

More libc wrappers are coming, but I do not see the big picture how they all will belong together yet. But this is a process happening outside of this repository - but at least we can detect breakage when updating the dependencies and respond accordingly.